sdk/core/System.ClientModel/tests/internal/Convenience/SSE/ServerSentEventReaderTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace System.ClientModel.Tests.Convenience;
+
+public class ServerSentEventReaderTests
+{
+    // TODO: Test both sync and async
+
+    [Test]
+    public async Task GetsEventsFromStream()
+    {
+        Stream contentStream = BinaryData.FromString(_mockContent).ToStream();
+        using ServerSentEventReader reader = new(contentStream);
+
+        List<ServerSentEvent> events = new();
+        ServerSentEvent? ssEvent = await reader.TryGetNextEventAsync();
+        while (ssEvent is not null)
+        {
+            events.Add(ssEvent.Value);
+            ssEvent = await reader.TryGetNextEventAsync();
+        }
+
+        Assert.AreEqual(events.Count, 3);
+
+        for (int i = 0; i < events.Count; i++)
+        {
+            ServerSentEvent sse = events[i];
+            Assert.IsTrue(sse.EventName.Span.SequenceEqual($"event.{i}".AsSpan()));
+            Assert.IsTrue(sse.Data.Span.SequenceEqual($"{{ \"id\": \"{i}\", \"object\": {i} }}".AsSpan()));
+        }
+
+        // TODO: Question - should this include the "done" event?  Probably yes?
+    }
+
+    #region Helpers
+
+    private string _mockContent = """
```

Comment:
TODO: put it in one place? Currently copied a lot.
Also: more robust set of test cases?